### DISCUSSION TITLE
New Releases of models should have ELI5 of what is recommended.

### DISCUSSION THREAD

**Discussion by shomilsaini** [immich-app/immich#11830] [Q&A]
So, I was looking into the new update and models and as a layman I don't understand anything in the release notes
So, as everyone I took to googling and still can't find any concrete answer. Finally, this post makes me feel connected, so referencing here.

Piggybacking of this: https://www.reddit.com/r/immich/comments/1esrf2p/comment/li82sn9/
```
Thanks, but I'm still unsure as to exactly why I'd choose each model. I have no interest in AI other than via immich, so I've paid it no attention, I have no idea how to read the charts in the PR's.

Ideally the docs would have some kind of table

model name	memory usage	notes
model_a	low (256mb)	fast but inaccurate
model_b	medium (512mb)	good all rounder
model_c	high (1024mb)	high quality but slower
```

Please do this moving forward, otherwise all the work that is being done is useless if no one can understand it. Thanks!

**Comment by radh21301** [comment]
I totally understand. I had no idea how to decide on the model, and I know nothing about ML.

Until we have something official in the docs, this discord thread might help:
https://discord.com/channels/979116623879368755/1273350673412521995

Something I found on discord:
![image](https://github.com/user-attachments/assets/a28d0cac-f9b6-4d85-9921-e30a0a1bcf25)

**Reply by shomilsaini** [reply]
Thanks, for the link. Although it is not a definitive answer, I will try to see if I can come to a conclusion for myself.

I feel like this is an ongoing issue because the new models will be developed now and again, and we need to know the relative performance and recommendations w.r.t VRAM requirements from immich team as time goes on.

If anyone would like to recommend something please feel free to do that. I am running P2000 with 5GB VRAM, server has 128GB RAM with Dual Xeon E5-2697v2 . I dont know what model will be best for me.

Thanks again!

**Reply by shomilsaini** [reply]
> I totally understand. I had no idea how to decide on the model, and I know nothing about ML.
> 
> Until we have something official in the docs, this discord thread might help: https://discord.com/channels/979116623879368755/1273350673412521995
> 
> Something I found on discord: ![image](https://private-user-images.githubusercontent.com/45341381/358356708-a28d0cac-f9b6-4d85-9921-e30a0a1bcf25.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3NDg5OTcsIm5iZiI6MTcyMzc0ODY5NywicGF0aCI6Ii80NTM0MTM4MS8zNTgzNTY3MDgtYTI4ZDBjYWMtZjliNi00ZDg1LTk5MjEtZTMwYTBhMWJjZjI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE1VDE5MDQ1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlYzE4ZDVmMjMwNGI0M2MyYzUzZjhmOTU1Mjc2YjJiYmFkOWMzOTQ5NDM4NGZlZjA0ODQ2ODMwZGZmODI5YmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-4WcgYPFFC68w_MGlca6HKQKlw_VgoTiFJ9tfqnGGs8)

I see you edited/updated the comment.

I actually changed to `ViT-B-16-SigLIP-384__webli` this morning after change docs praised highly. Furthermore, I was running `XLM-Roberta-Large-Vit-B-32` before. Looks like it was the correct move, or maybe I could have used something better. I am not sure with my limited knowledge. (You see the issue with this. 🥲 hint: people can't figure out what is recommended/appropriate for their hardware)

I am still running antelopev2 for facial recognition.

Thanks

**Reply by radh21301** [reply]
I think the general idea is: The more MACs the model has, the slower it is going to be. But how slow depends on the hardware. So, use the largest model (the one that has the best efficiency/average score), if you have enough RAM and a decently powerful setup -> Observe the 'speed', the time it takes to run the 'smart search job' and regualr contextual search once the job is completed -> if that is seems okay to you, then why not stick with it right. Else, try a smaller mode..keep stepping down till you are happy.

I dont have a powerful machine, so I went with 'ViT-B-32__laion2b_e16'

**Reply by raisinbear** [reply]
> I think the general idea is: The more MACs the model has, the slower it is going to be. But how slow depends on the hardware. So, use the largest model (the one that has the best efficiency/average score), if you have enough RAM and a decently powerful setup -> Observe the 'speed', the time it takes to run the 'smart search job' and regualr contextual search once the job is completed -> if that is seems okay to you, then why not stick with it right. Else, try a smaller mode..keep stepping down till you are happy.
> 
> I dont have a powerful machine, so I went with 'ViT-B-32__laion2b_e16'

Ok that’s interesting, thanks. Have been searching for info on the performance implications for a while. I already asked this in the release discussion, but this is probably the better place for it:

Does anyone happen to know whether this slow down is for classification, search or both? I wouldn’t mind longer processing times after upload, but don’t really want to slow down search by a factor of 3 (MACs of the 16-SigLIP__webli model vs 32__openai) and it’s not that straightforward to check, either, without rerunning smart search on all assets 😅

**Reply by radh21301** [reply]
It affects both the initial processing and duration for each and every search as well. So yeah, higher the MACs, the longer it takes to get results after a search. That was my exact concern as well. So, I stuck with a smaller model.
Refer to this thread for more info:

https://github.com/immich-app/immich/discussions/11862

**Reply by raisinbear** [reply]
> It affects both the initial processing and duration for each and every search as well. So yeah, higher the MACs, the longer it takes to get results after a search. That was my exact concern as well. So, I stuck with a smaller model. Refer to this thread for more info:
> 
> #11862

Cool, thanks for pointing me there! 👍
